client/add/pokemon.ctrl.js

```diff
@@ -23,6 +23,10 @@ module.exports = class PokemonAdd {
     this.lines = [];
   }
 
+  hasPublicPokemon () {
+    return Boolean(this.lines.find((line) => line.visibility === 'public'));
```

Comment:
Nit: instead of `Boolean(this.lines.find(func))`, it's simpler to use `this.lines.some(func)`.